test/system/authentication_test.rb

```diff
@@ -46,6 +46,8 @@ class AuthenticationTest < ApplicationSystemTestCase
     end
 
     assert_not authentication.succeeded?
+    assert_equal "rejected", authentication.events&.last&.action
+    assert_equal "expired", authentication.events&.last&.details&.dig("reason")
```

Comment:
Might wanna add these assertions for the invalid link test too.

Reply:
With an invalid link, we aren't able to match the request to an Authentication record. So, there's no Authentication for us to record Events on to